cocos/2d/CCSprite.cpp

```diff
@@ -427,48 +428,103 @@ void Sprite::updatePoly()
 {
     if (_numberOfSlices == 1) {
         setTextureCoords(_rect, &_quad);
-        const Rect copyRect(0, 0, _rect.size.width * _strechFactor.x, _rect.size.height * _strechFactor.y);
-        setVertexCoords(copyRect, _rect.size, &_quad);
+        Rect copyRect;
+        if (_strechEnabled) {
+            copyRect = Rect(0, 0, _rect.size.width * _strechFactor.x, _rect.size.height * _strechFactor.y);
+        } else {
+            copyRect = Rect((_contentSize.width - _originalContentSize.width) / 2.0f,
+                            (_contentSize.height - _originalContentSize.height) / 2.0f,
```

Comment:
kind of... I have to do that calculation in two places. but it would be difficult to avoid that.
no problem in having a `_contentSize < _originalSize`